Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/NodeTest.java

```diff
@@ -0,0 +1,1134 @@
+            Node node = storage.createNode(Mods.NAME);
+            node.add(storage.createNode("http://names.example/alice"));
+            node.add(storage.createNode("http://names.example/bob"));
+            node.add(storage.createNode("http://names.example/charlie"));
+
+            Node condition = storage.createNode().put(RDF.toURL(2), storage.createNode("http://names.example/bob"));
+
+            assertThat(node.matches(condition), is(true));
+        }
+    }
+
+    /**
+     * Tests {@code matches(ObjectType)} for a non-matching candidate.
+     */
+    @Test
+    public void testMatchesNot() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Node node = storage.createNode(Mods.NAME);
+            node.add(storage.createNode("http://names.example/alice"));
+            node.add(storage.createNode("http://names.example/bob"));
+            node.add(storage.createNode("http://names.example/charlie"));
+
+            Node condition = storage.createNode(Mods.NAME).put(RDF.toURL(9),
+                storage.createNode("http://names.example/alice"));
+
+            assertThat(node.matches(condition), is(false));
+        }
+    }
+
+    /**
+     * Tests {@code putAll(IdentifiableNode, Collection<? extends ObjectType>)}.
+     */
+    @Test
+    public void testPutAllIdentifiableNodeSet() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            List<ObjectType> nodes = new ArrayList<>();
+            nodes.add(storage.createNodeReference("http://names.example/alice"));
+            nodes.add(storage.createNodeReference("http://names.example/bob"));
+            nodes.add(storage.createNodeReference("http://names.example/charlie"));
+
+            Node node = storage.createNode();
+            node.putAll(Mods.NAME, nodes);
+
+            assertThat(node.get(Mods.NAME).countUntil(Long.MAX_VALUE), is((long) 3));
+        }
+    }
+
+    /**
+     * Tests {@code putAll(String, Collection<? extends ObjectType>)}.
+     */
+    @Test
+    public void testPutAllStringCollection() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            List<ObjectType> nodes = new ArrayList<>();
+            nodes.add(storage.createNodeReference("http://names.example/alice"));
+            nodes.add(storage.createNodeReference("http://names.example/bob"));
+            nodes.add(storage.createNodeReference("http://names.example/charlie"));
+
+            Node node = storage.createNode();
+            node.putAll(Mods.NAME.getIdentifier(), nodes);
+
+            assertThat(node.get(Mods.NAME).countUntil(Long.MAX_VALUE), is((long) 3));
+        }
+    }
+
+    /**
+     * Tests {@code put(IdentifiableNode, ObjectType)}.
+     */
+    @Test
+    public void testPutNodeReferenceObjectType() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNode().put(Mods.NAME, storage.createLeaf("Wilhelm Busch", "de"));
```

Comment:
What did you testing here?

Reply:
Tests that put method works (i.e. does not throw an exception). This seems trivial for the current implementation, but the test is written on the interface and if extended to a future implementation of the interface, maybe it is not that trivial.